contracts/src/v0.8/functions/dev/0_0_0/FunctionsOracle.sol

```diff
@@ -255,15 +255,17 @@ contract FunctionsOracle is Initializable, IFunctionsOracle, OCR2BaseUpgradeable
       returns (IFunctionsBillingRegistry.FulfillResult result) {
         if (result == IFunctionsBillingRegistry.FulfillResult.USER_SUCCESS) {
           emit OracleResponse(requestIds[i]);
+          emit ResponseTransmitted(requestIds[i], transmitter);
         } else if (result == IFunctionsBillingRegistry.FulfillResult.USER_ERROR) {
           emit UserCallbackError(requestIds[i], "error in callback");
+          emit ResponseTransmitted(requestIds[i], transmitter);
         } else if (result == IFunctionsBillingRegistry.FulfillResult.INVALID_REQUEST_ID) {
           emit InvalidRequestID(requestIds[i]);
         }
       } catch (bytes memory reason) {
         emit UserCallbackRawError(requestIds[i], reason);
+        emit ResponseTransmitted(requestIds[i], transmitter);
       }
-      emit ResponseTransmitted(requestIds[i], transmitter);
     }
```

Comment:
FYI: I tried do this with a local boolean variable, e.g. `shallEmitResponseTransmitted`, but the compiler yield error: "stack depth is too deep", because `fulfillAndBill` function likely takes all stack capacity.

Reply:
The current approach is more readable IMO. Stack depth error sounds suspicious, maybe @justinkaseman can explain.